hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -696,7 +702,7 @@ public void run() {
             // When retrying, we must reset lastHeartbeat, otherwise InvocationMonitor will see the old value
             // and falsely conclude that nothing has been done about this operation for a long time.
             lastHeartbeatMillis = 0;
-            doInvoke(false);
+            doInvoke(true);
```

Comment:
Generic operations can be executed on calling thread directly, if not async. But invocation monitor scheduler can't execute operations.